drivers/sensor/adi/adxl34x/adxl34x_rtio.c

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright (c) 2024 Chaim Zax
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/sensor/adxl34x.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/device_runtime.h>
+#include <zephyr/rtio/rtio.h>
+#include <zephyr/sys/util.h>
+
+#include "adxl34x_reg.h"
+#include "adxl34x_private.h"
+#include "adxl34x_rtio.h"
+#include "adxl34x_decoder.h"
+#include "adxl34x_trigger.h"
+
+LOG_MODULE_DECLARE(adxl34x, CONFIG_SENSOR_LOG_LEVEL);
+
+/**
+ * Fetch a single sample from the sensor
+ *
+ * @param[in] dev Pointer to the sensor device
+ * @param[out] rx_buf Pointer to store the result
+ * @param[in] buf_size Size of the @p rx_buf buffer in bytes
+ * @return 0 if successful, negative errno code if failure.
+ */
+static int adxl34x_rtio_sample_fetch(const struct device *dev, uint8_t *rx_buf, uint8_t buf_size)
+{
+	const struct adxl34x_dev_config *config = dev->config;
+	int rc;
+
+	/* Read accel x, y and z values. */
+	rc = config->bus_read_buf(dev, ADXL34X_REG_DATA, rx_buf, buf_size);
```

Comment:
This looks to be a blocking read, any particular reason not to trigger an async read here?

Imagine wishing to start the read periodically directly from a timer interrupt for example.

Reply:
No particular reason. I had some issues finding a good reference driver and started with a relative simple approach (using blocking calls). The icm42688 driver is the only one using the RTIO system to do the actual I2C commands as well for streaming data, which does use this async mechanism (as far as I can tell). I can do the same here.